app/uk/gov/hmrc/testuser/services/TestUserService.scala

```diff
@@ -29,15 +33,25 @@ trait TestUserService {
 
   def createTestIndividual() = {
     val individual = generator.generateTestIndividual()
-    val hashedPassword =  passwordService.hash(individual.password)
+    val hashedPassword = passwordService.hash(individual.password)
     testUserRepository.createUser(individual.copy(password = hashedPassword)) map (_ => individual)
   }
 
   def createTestOrganisation() = {
     val organisation = generator.generateTestOrganisation()
-    val hashedPassword =  passwordService.hash(organisation.password)
+    val hashedPassword = passwordService.hash(organisation.password)
     testUserRepository.createUser(organisation.copy(password = hashedPassword)) map (_ => organisation)
   }
+
+  def authenticate(username: String, password: String) =
+    testUserRepository.fetchByUsername(username).map {
+      case None => Unauthorized(Json.toJson(ErrorResponse.usernameNotFoundError(username)))
+      case Some(ind @ TestIndividual(`username`, hashedPass, _, _, _)) if passwordService.validate(password, hashedPass) =>
+        Ok(Json.toJson(TestIndividualResponse.from(ind)).toString())
+      case Some(org @ TestOrganisation(`username`, hashedPass, _, _, _, _, _)) if passwordService.validate(password, hashedPass) =>
+        Ok(Json.toJson(TestOrganisationResponse.from(org)).toString())
+      case _ => Unauthorized(Json.toJson(ErrorResponse.wrongPasswordError(username)))
+    }
```

Comment:
The http statuses and json responses should stay in the controller so I think we can clean it up with something like this:
  def authenticate(username: String, password: String) =
    testUserRepository.fetchByUsername(username).map {
      case None => throw new UserNotFoundException(username)
      case Some(e: TestUser) if passwordService.validate(password, e.password) => e
      case _ => throw new InvalidCredentials(username)
    }